### DISCUSSION TITLE
nginx problem after upgrading to php 8.3 (permisison denied for fpm)

### DISCUSSION THREAD

**Discussion by bandiba** [firefly-iii/firefly-iii#9119] [Firefly III]
Hi,

Unluckily I had a lower than 8.3 version of php when installed Firefly. I was warned when tried to use the data importer. So I upgraded to php 8.3. I also reinstalled/restarted nginx. However now nginx gives me 502 bad gateway. 
The nginx error log has entries like this:

"2024/08/04 11:46:20 [crit] 63272#63272: *7 connect() to unix:/run/php/php8.3-fpm.sock failed (13: Permission denied) while connecting to upstream, client: 192.168.0.101, server: localhost, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php8.3-fpm.sock:", host: "xxxx.yyyy.lan""

Any ideas pls what I could try?

Thanks.

**Comment by JC5** [comment]
Mmm, permission denied on the FPM socket, that's weird. Is the configuration the same otherwise as the 8.2 was?

**Comment by bandiba** [comment]
Well, _I think_ it is... While obviously I'm not mastering php neither nginx. I recall there was something wrong with the nginx.conf, that I assumed I handled based on some hints I found in slashdot, or elsewhere.
For the php upgrade I followed an 8.2 to 8.3 ubuntu guide (I have ubuntu 22.04), I recall it transfers configs too.
I'm not using php or nginx for anything else than firefly, so I'm ready to remove stuff and start from scratch, if that may help. Although not sure what is the best way to do it.

**Reply by JC5** [reply]
Mmm, did you try the basics, like rebooting your server? Sometimes that helps, weirdly enough.

Is there something on Google re: "access denied php fpm"?

**Reply by bandiba** [reply]
_Updated_ on 2024.08.22: problem sorted out.

Hi, I can report some success, after diligently googling, fixing and rebooting. The problems were mostly with the nginx.conf and firefly-iii.conf.

Now I can access Firefly, which is great! Thanks.

Next is the data importer, which I installed, _and just found the config mistake, so now data importer also loads nicely!_

Thanks!

**Reply by JC5** [reply]
Nice work! Sorry I couldn't respond earlier.

**Reply by bandiba** [reply]
No worries, it was a very educative exercise. Overall, congratulation for the project, you clearly invested a lot of work and a lot of thought. 

If you allow me a small suggestion for the [importer how-to](https://docs.firefly-iii.org/how-to/data-importer/installation/self-managed/).
It says _"In the directory where you just unzipped Firefly III you will find a .env.example file. Rename or copy it to .env."_ 
I'm pretty sure this should be _data-importer_ instead of _Firefly III_. Probably copy-paste mistake.

And a humble ask: I would badly need the reference currency option, so that the reports can show a single (reference) currency only. I will comment on the related discussion too.

**Reply by JC5** [reply]
Nice, I'll update the text :)
